cms/djangoapps/modulestore_migrator/rest_api/v1/views.py

```diff
@@ -0,0 +1,137 @@
+                "completed_steps": 11,
+                "total_steps": 11,
+                "attempts": 1,
+                "created": "2025-05-14T22:24:37.048539Z",
+                "modified": "2025-05-14T22:24:59.128068Z",
+                "artifacts": [],
+                "parameters": {
+                    "source": "course-v1:OpenedX+DemoX+DemoCourse",
+                    "composition_level": "unit",
+                    "repeat_handling_strategy": "update",
+                    "preserve_url_slugs": true
+                }
+            }
+
+    GET /api/modulestore_migrator/v1/migrations/<uuid>/
+        Get the status of the import task.
+
+        Example response:
+            {
+                "name": "migrate_from_modulestore",
+                "state": "Importing staged content structure",
+                "state_text": "Importing staged content structure",
+                "completed_steps": 6,
+                "total_steps": 11,
+                "attempts": 1,
+                "created": "2025-05-14T22:24:37.048539Z",
+                "modified": "2025-05-14T22:24:59.128068Z",
+                "artifacts": [],
+                "parameters": {
+                    "source": "course-v1:OpenedX+DemoX+DemoCourse2",
+                    "composition_level": "component",
+                    "repeat_handling_strategy": "skip",
+                    "preserve_url_slugs": false
+                }
+            }
+    """
+
+    permission_classes = (IsAdminUser,)
+    authentication_classes = (
+        BearerAuthenticationAllowInactiveUser,
+        JwtAuthentication,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    serializer_class = StatusWithModulestoreMigrationSerializer
+
+    def get_queryset(self):
+        """
+        Override the default queryset to filter by the import event and user.
+        """
+        return StatusViewSet.queryset.filter(modulestoremigration__isnull=False, user=self.request.user)
+
+    def create(self, request, *args, **kwargs):
+        """
+        Handle the import task creation.
+        """
+
+        serializer_data = ModulestoreMigrationSerializer(data=request.data)
+        serializer_data.is_valid(raise_exception=True)
+        validated_data = serializer_data.validated_data
+
+        try:
+            task = start_migration_to_library(
+                user=request.user,
+                source_key=validated_data['source'],
+                target_library_key=validated_data['target'],
+                target_collection_slug=validated_data['target_collection_slug'],
+                composition_level=validated_data['composition_level'],
+                repeat_handling_strategy=validated_data['repeat_handling_strategy'],
+                preserve_url_slugs=validated_data['preserve_url_slugs'],
+                forward_source_to_target=False,  # @@TODO - Set to False for now. Explain this better.
```

Comment:
@cmltaWt0 can you set this from the serializer data and remove the comment?

```suggestion
                forward_source_to_target=False,  # @@TODO - Set to False for now. Explain this better.
```

Reply:
@kdmccormick Do we need this param in response data?
```json
{
    "artifacts": [],
    "parameters": {
        "source": "ModulestoreSource('library-v1:RG+lib-legacy')",
        "target": "lib:RG:test19",
        "composition_level": "component",
        "repeat_handling_strategy": "update",
        "preserve_url_slugs": false,
        "forward_source_to_target": true
    }
}
```

Asking because there is no such field and I need either to pass this param directly to response or implement custom logic to get the non existent field.

Right now it can be as simple as this line in view:
```python
serializer.data["parameters"]["forward_source_to_target"] = validated_data['forward_source_to_target']
```

Reply:
ah good call out.

We need this to be returned from the GET as well, so that would mean adding it to the serializer with custom logic, right? If so, please do that.

Reply:
^ Reminder on this.

I think the logic could be something like this:

```python
def get_forward_source_to_target(self, obj: ModulestoreMigration):
    return obj.id == obj.source.forwarded_id
```